design/layer2-bind-interfaces.md

```diff
@@ -0,0 +1,243 @@
+  name: example-advertisement4
+  namespace: metallb-system
+spec:
+  ipAddressPools:
+    - pool2
+  interfaces:
+    - veth0
+```
+
+### Avoid spamming ARP to veth interfaces
+
+Specify that all VIPs are advertised from all physical interfaces, except the virtual interfaces that belong to the pods 
+in the form of veth pair. For example, if each node in the cluster has 3 physical interfaces: eth0, eth1, eth2:
+```yaml
+apiVersion: metallb.io/v1beta1
+kind: L2Advertisement
+metadata:
+name: example-advertisement5
+namespace: metallb-system
+spec:
+  interfaces:
+    - eth0
+    - eth1
+    - eth2
+```
+
+### Announcing the VIP from specific interfaces only from a subset of nodes
+
+By mixing `nodeSelectors` and `interfaces`, we can consider announcing pool1 from hostA and hostB, using the selected 
+host's ens18 interface:
+```yaml
+apiVersion: metallb.io/v1beta1
+kind: L2Advertisement
+metadata:
+  name: example-advertisement6
+  namespace: metallb-system
+spec:
+  ipaddresspools:
+    - pool1
+  nodeSelector:
+    - matchExpression:
+      - key: kubernetes.io/hostname
+        operator: In
+        values: [hostA, hostB]
+  interfaces:
+    - ens18
+```
+
+### Specify different interfaces for different nodes to announce VIP
+
+In a heterogeneous cluster, the interfaces of each node are not completely consistent. We need to specify different interfaces for different nodes.
+For example, when announcing from "worker" nodes use only eno1 interface, but only enp3s5 and vlan6 on "gateway" nodes.
+```yaml
+apiVersion: metallb.io/v1beta1
+kind: L2Advertisement
+metadata:
+  name: example-advertisement7
+  namespace: metallb-system
+spec:
+  ipaddresspools:
+    - pool1
+  nodeSelector:
+    - role: worker
+  interfaces:
+    - eno1
+---
+apiVersion: metallb.io/v1beta1
+kind: L2Advertisement
+metadata:
+  name: example-advertisement8
+  namespace: metallb-system
+spec:
+  ipaddresspools:
+    - pool1
+  nodeSelector:
+    - networkRole: gateway
+  interfaces:
+    - enp3s5
+    - vlan6
+```
+
+#### Note: Multiple L2Advertisements are combined by taking the union
+
+For example:
+```yaml
+apiVersion: metallb.io/v1beta1
+kind: L2Advertisement
+metadata:
+  name: example-advertisement9
+  namespace: metallb-system
+spec:
+  interfaces:
+    - eno1
+---
+apiVersion: metallb.io/v1beta1
+kind: L2Advertisement
+metadata:
+  name: example-advertisement10
+  namespace: metallb-system
+spec:
+  ipaddresspools:
+    - pool1
+  nodeSelector:
+    - kubernetes.io/hostname: hostB
+  interfaces:
+    - ens18
+```
+The above YAML indicates that MetalLB should advertise the VIPs of all IPAddressPools including pool1 from the interface 
+eno1 of all nodes, and also advertise the VIPs in pool1 from ens18 of hostB. In other words, if MetalLB chooses hostB to 
+announce the VIP of pool1, the Speaker should announce the VIP from the interfaces ens18 and eno1; if it chooses other nodes, 
+the Speaker should announce the VIP only from the interface eno1.
+
+### Test Plan
```

Comment:
Can you elaborate more about what/how you're going to test?
like this for example: https://github.com/metallb/metallb/blob/main/design/0001-frr.md#test-coverage

Reply:
> Can you elaborate more about what/how you're going to test? like this for example: https://github.com/metallb/metallb/blob/main/design/0001-frr.md#test-coverage

@qingwusunny ping, not sure you missed this comment

Reply:
Sorry, I have been busy recently. And I will reply later.

Reply:
No rush at all! I wanted just to make sure you did not miss it, thanks again!